plugins/docker-options/functions

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+[[ " docker-options docker-options:add docker-options:remove help docker-options:help " == *" $1 "* ]] || exit $DOKKU_NOT_IMPLEMENTED_EXIT
+set -eo pipefail; [[ $DOKKU_TRACE ]] && set -x
+source "$PLUGIN_CORE_AVAILABLE_PATH/common/functions"
+
+get_phases() {
+  local passed_phases_list="$1"
+  local phases_allowed=$(sed -e 's/ /\|/g' <<< "${AVAILABLE_PHASES[@]}")
+  local phase
+  local passed_phases
+  if [[ -n "$passed_phases_list" ]]; then
+    IFS=',' read -ra passed_phases <<< "$passed_phases_list"
+    for phase in "${passed_phases[@]}"; do
+      [[ "$phase" = @($phases_allowed) ]] || dokku_log_fail "Phase(s) must be one of [${AVAILABLE_PHASES[*]}]"
+    done
+    echo "${passed_phases[@]}"
+  fi
+}
+
+create_phase_file_if_required() {
+  local phase_file_path="$1"
+  [[ -f "$phase_file_path" ]] || {
+    touch "$phase_file_path"
+  }
+}
+
+display_phase_options() {
+  local phase="$1"
+  local phase_file_path="$2"
+  echo "${phase^} options:"
+  sed -e 's/^/    /' "$phase_file_path"
+}
+
+display_all_phases_options() {
```

Comment:
Problem is `PHASE_FILE_PATH` is used in a for loop that defines `phase` which is a component of `PHASE_FILE_PATH`. I think moving towards a more readable version of this plugin would be best.